class ResponseNotOkException(Exception):
    """
    Custom exception raised when an API response is not successful.

    This exception is used to indicate that the response from the Outline VPN server was unsuccessful, 
    typically when the HTTP status code is 300 or higher, indicating an error. It carries the error message 
    generated by the `_get_error_message` function, providing additional context about the failure.

    Args:
        *args: Variable length argument list that can include the error message(s) to be passed to the 
               base `Exception` class.
    """
    def __init__(self, *args):
        super().__init__(*args)